### ISSUE TITLE
type 'Null' is not a subtype of type 'String' in type cast ( metamask rejection reason error )

### ISSUE THREAD

**Issue by wufd** [WalletConnect/WalletConnectFlutterV2#118]
Use metamask to sign the transaction, when the user clicks to reject, there is a problem with the parsing parameters of walletconnect  

```
type 'Null' is not a subtype of type 'String' in type cast
#0      _$JsonRpcErrorFromJson (package:walletconnect_flutter_v2/apis/models/json_rpc_error.g.dart:11:32)
#1      new JsonRpcError.fromJson (package:walletconnect_flutter_v2/apis/models/json_rpc_error.dart:27:7)
#2      _$JsonRpcResponseFromJson (package:walletconnect_flutter_v2/apis/models/json_rpc_response.g.dart:18:26)
#3      new JsonRpcResponse.fromJson (package:walletconnect_flutter_v2/apis/models/json_rpc_response.dart:21:7)
#4      Pairing._onMessageEvent (package:walletconnect_flutter_v2/apis/core/pairing/pairing.dart:655:40)
                   <asynchronous suspension>
```

**Comment by nqhhdev** [comment]
Can you show me your code for me? We are discus now

**Comment by wufd** [comment]
you can try this.   to Metamask  click  'reject' button
```
await connector.request(
      topic: topic,
      chainId: eipChain,
      request: SessionRequestParams(
        method: 'eth_sendTransaction',
        params: [
          {
            'from': from,
            'data': data,
            'to': to,
          }
        ],
      ),
    );
```

**Comment by nqhhdev** [comment]
can you show me `connector`  func now?

**Comment by wufd** [comment]
use 
```
connector = await Web3App.createInstance   
# or  
connector = await SignClient.createInstance( xxxx )
```

**Comment by nqhhdev** [comment]
I need to try to connect with MM. Can you share it for me

**Comment by wufd** [comment]
you can try the lib-example, has same issue:
https://github.com/WalletConnect/WalletConnectFlutterV2/tree/master/example/dapp

**Comment by wufd** [comment]
similar issues :
```
Unhandled Exception: type 'Null' is not a subtype of type 'int' in type cast
#0      _$JsonRpcErrorFromJson (package:walletconnect_flutter_v2/apis/models/json_rpc_error.g.dart:10:26)
#1      new JsonRpcError.fromJson (package:walletconnect_flutter_v2/apis/models/json_rpc_error.dart:27:7)
#2      _$JsonRpcResponseFromJson (package:walletconnect_flutter_v2/apis/models/json_rpc_response.g.dart:18:26)
#3      new JsonRpcResponse.fromJson (package:walletconnect_flutter_v2/apis/models/json_rpc_response.dart:21:7)
#4      Pairing._onMessageEvent (package:walletconnect_flutter_v2/apis/core/pairing/pairing.dart:655:40)
        <asynchronous suspension>
```

**Comment by Luzzotica** [comment]
Is this the most recent version of MetaMask?

It looks like the `code` value for the error is null, which is throwing the error.

That code should ever be null in a JSON RPC request/response/error.

**Comment by wufd** [comment]
is lastest metamask (iOS 7.0.1).
So the parameter type of the model should be optional? or has a default value?

**Comment by nqhhdev** [comment]
@wufd You can follow my ticket 
https://github.com/WalletConnect/WalletConnectFlutterV2/issues/113#issuecomment-1614343277

**Comment by Luzzotica** [comment]
I was wrong, the `message` was null.
This should also not be possible.

https://www.jsonrpc.org/specification#error_object

I have made the message optional to resolve this problem, but it's still a weird issue.

I'll close this issue when I push out the new version.

**Comment by nqhhdev** [comment]
@Luzzotica Any update? 

If user clicks on reject, How do I go back to the app?

**Comment by Luzzotica** [comment]
The null error is resolved with v2.0.13

I have no control over metamask and how they manage redirects. If tapping on reject in MetaMask doesn't redirect back to the dapp, you must take that up with them.

**Comment by rkovrigin** [comment]
There is a problem with v2.0.13.
It requires http v1.1.3, and many other libraries depend on lower versions of http, so, that breaks many dependencies =(

**Comment by bobwith2bees** [comment]
> The null error is resolved with v2.0.13
> 
> I have no control over metamask and how they manage redirects. If tapping on reject in MetaMask doesn't redirect back to the dapp, you must take that up with them.


@Luzzotica -  
Just a note:   The protocol peer/pairing metadata contains an optional Redirect where an app can specify the native or universal link the peer should use to open to their app.   

Since this is an optional field, I have yet to see anyone populate it.   My initial design planned on using it, but in practice I just had to store all the icons and launch URI's from the initial web3modal selection.  

@nqhhdev  - You could submit a bug/feature request for Metamask to support this peer Redirect if they do not honor it.

**Comment by rkovrigin** [comment]
Tried to solve dependencies, but now I am using v2.0.13, the issue is like that

#0      _$$_JsonRpcErrorFromJson (package:walletconnect_flutter_v2/apis/models/json_rpc_error.g.dart:11:26)
#1      new _$_JsonRpcError.fromJson (package:walletconnect_flutter_v2/apis/models/json_rpc_error.freezed.dart:115:7)
#2      _$JsonRpcErrorFromJson (package:walletconnect_flutter_v2/apis/models/json_rpc_error.freezed.dart:18:24)
#3      new JsonRpcError.fromJson (package:walletconnect_flutter_v2/apis/models/json_rpc_error.dart:26:7)
#4      _$$_JsonRpcResponseFromJson (package:walletconnect_flutter_v2/apis/models/json_rpc_response.g.dart:18:26)
#5      new _$_JsonRpcResponse.fromJson (package:walletconnect_flutter_v2/apis/models/json_rpc_response.freezed.dart:157:7)
#6      _$JsonRpcResponseFromJson (package:walletconnect_flutter_v2/apis/models/json_rpc_response.freezed.dart:19:10)
#7      new JsonRpcResponse.fromJson (package:walletconnect_flutter_v2/apis/models/json_rpc_response.dart:18:7)
#8      Pairing._onMessageEvent (package:walletconnect_flutter_v2/apis/core/pairing/pairing.dart:656:40)
